internal/credential/credential_libraries.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// CredentialLibraryRepository defines the interface expected
+// to get the total number of credential libraries.
+type CredentialLibraryRepository interface {
+	GetTotalCredentialLibraries(context.Context) (int, error)
+	GetTotalSSHCertificateCredentialLibraries(context.Context) (int, error)
+	ListDeletedCredentialLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListDeletedSSHCertificateCredentialLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+}
+
+// GetTotalCredentialLibraries gets the total number of credential libraries across all types
+func GetTotalCredentialLibraries(ctx context.Context, repo CredentialLibraryRepository) (int, error) {
+	numGenericLibs, err := repo.GetTotalCredentialLibraries(ctx)
+	if err != nil {
+		return 0, nil
+	}
+	numSSHCertLibs, err := repo.GetTotalSSHCertificateCredentialLibraries(ctx)
+	if err != nil {
+		return 0, nil
+	}
+	return numGenericLibs + numSSHCertLibs, nil
+}
+
+// ListDeletedCredentialLibraryIds lists all deleted credential library IDs across all types
+func ListDeletedCredentialLibraryIds(ctx context.Context, rw db.Writer, repo CredentialLibraryRepository, since time.Time) ([]string, error) {
+	const op = "credential.ListDeletedCredentialLibraryIds"
+	var deletedIds []string
+	_, err := rw.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{}, func(r db.Reader, w db.Writer) error {
+		deletedLibIds, err := repo.ListDeletedCredentialLibraryIds(ctx, since, WithReaderWriter(r, w))
+		if err != nil {
+			return err
+		}
+		deletedSSHCertLibIds, err := repo.ListDeletedSSHCertificateCredentialLibraryIds(ctx, since, WithReaderWriter(r, w))
+		if err != nil {
+			return err
+		}
+		deletedIds = append(deletedLibIds, deletedSSHCertLibIds...)
+		// TODO: Get transaction timestamp too
```

Comment:
In a future PR I will add the transaction timestamp to the db.Writer and it'll be available here and returned from this function.